lms/djangoapps/certificates/tasks.py

```diff
@@ -18,11 +19,16 @@ class _BaseCertificateTask(PersistOnFailureTask, LoggedTask):  # pylint: disable
     abstract = True
 
 
-@task(base=_BaseCertificateTask)
-def generate_certificate(**kwargs):
+@task(base=_BaseCertificateTask, bind=True, default_retry_delay=30, max_retries=2)
+def generate_certificate(self, **kwargs):
     """
     Generates a certificate for a single user.
     """
     student = User.objects.get(id=kwargs.pop('student'))
     course_key = CourseKey.from_string(kwargs.pop('course_key'))
+    expected_verification_status = kwargs.pop('verification', None)
+    if expected_verification_status:
```

Comment:
You'll want to add a test in `certificates/test_tasks.py` to cover this if-block.  There's a factory in `lms/djangoapps/verify_student/tests/factories.py` that will help you create a test photo verification object.